acceptance/tests/agent/enforce_runtimeout.rb

```diff
@@ -0,0 +1,59 @@
+require 'beaker/host_prebuilt_steps'
+
+test_name 'enforce runtimeout' do
+  extend Beaker::HostPrebuiltSteps
+
+  teardown do
+    stop_agent_on(agents)
+    on(master, 'rm -f /etc/puppetlabs/code/environments/production/manifests/site.pp')
+  end
+
+  step 'set up puppet code for putting agents to sleep' do
+    copy_file_to_remote(master, '/etc/puppetlabs/code/environments/production/manifests/site.pp', <<-EOF)
```

Comment:
`copy_file_to_remote` I think is a beaker internal helper that it can use before it's loaded the whole DSL during setup. I think for the same behavior in tests `create_remote_file` can be used and comes within the DSL (ie you can remove the require/extend above). Here's an example [usage](https://github.com/puppetlabs/puppet/blob/master/acceptance/tests/resource/exec/should_accept_large_output.rb#L16).